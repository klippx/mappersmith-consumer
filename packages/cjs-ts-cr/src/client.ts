const mappersmith = require("mappersmith");
const { default: forge, configs } = mappersmith;
const FetchGateway = require("mappersmith/gateway/fetch");
const _HTTPGateway = require("mappersmith/gateway/http");
const _MockGateway = require("mappersmith/gateway/mock");
const _XHRGateway = require("mappersmith/gateway/xhr");

const _RetryMiddlewareV1 = require("mappersmith/middleware/retry");
const _RetryMiddlewareV1Alt = require("mappersmith/middleware/retry/v1");
const RetryMiddleware = require("mappersmith/middleware/retry/v2");
const BasicAuth = require("mappersmith/middleware/basic-auth");
const CSRF = require("mappersmith/middleware/csrf");
const Duration = require("mappersmith/middleware/duration");
const EncodeJSON = require("mappersmith/middleware/encode-json");
const ErrorHandler = require("mappersmith/middleware/global-error-handler");
const Log = require("mappersmith/middleware/log");
const Timeout = require("mappersmith/middleware/timeout");

configs.gateway = FetchGateway.default;

export const github = forge({
  clientId: "github",
  host: "https://www.githubstatus.com",
  middleware: [
    RetryMiddleware.default(),
    BasicAuth.default({
      username: "username",
      password: "password",
    }),
    CSRF.default(),
    Duration.default,
    EncodeJSON.default,
    ErrorHandler.default,
    Log.default,
    Timeout.default(1000),
  ],
  resources: {
    Status: {
      current: { path: "/api/v2/status.json" },
      summary: { path: "/api/v2/summary.json" },
      components: { path: "/api/v2/components.json" },
    },
  },
});
