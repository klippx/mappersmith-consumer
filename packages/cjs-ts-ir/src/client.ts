import mappersmith = require("mappersmith");
const { default: forge, Response, configs, version } = mappersmith;
import FetchGateway = require("mappersmith/gateway/fetch");
import _HTTPGateway = require("mappersmith/gateway/http");
import _MockGateway = require("mappersmith/gateway/mock");
// import _XHRGateway = require("mappersmith/gateway/xhr");
import TimeoutError = require("mappersmith/gateway/timeout-error");
const { isTimeoutError, createTimeoutError } = TimeoutError;

// import _RetryMiddlewareV1 = require("mappersmith/middleware/retry");
import _RetryMiddlewareV1Alt = require("mappersmith/middleware/retry/v1");
import RetryMiddleware = require("mappersmith/middleware/retry/v2");
import BasicAuth = require("mappersmith/middlewares/basic-auth");
import CSRF = require("mappersmith/middleware/csrf");
import Duration = require("mappersmith/middleware/duration");
import EncodeJSON = require("mappersmith/middleware/encode-json");
import ErrorHandler = require("mappersmith/middleware/global-error-handler");
import Log = require("mappersmith/middleware/log");
import Timeout = require("mappersmith/middleware/timeout");

configs.gateway = FetchGateway.default;

export const github = forge({
  clientId: "github",
  middleware: [
    RetryMiddleware.default(),
    BasicAuth.default({
      username: "username",
      password: "password",
    }),
    CSRF.default(),
    Duration.default,
    EncodeJSON.default,
    ErrorHandler.default,
    Log.default,
    Timeout.default(1000),
  ],
  host: "https://www.githubstatus.com",
  resources: {
    Status: {
      current: { path: "/api/v2/status.json" },
      summary: { path: "/api/v2/summary.json" },
      components: { path: "/api/v2/components.json" },
    },
  },
});
