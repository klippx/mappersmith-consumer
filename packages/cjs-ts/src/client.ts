import forge, { configs } from "mappersmith";
import fetch from "node-fetch";
import FetchGateway from "mappersmith/gateway/fetch";
import _HTTPGateway from "mappersmith/gateway/http";
import _MockGateway from "mappersmith/gateway/mock";
import _XHRGateway from "mappersmith/gateway/xhr";

import _RetryMiddlewareV1 from "mappersmith/middleware/retry";
import _RetryMiddlewareV1Alt from "mappersmith/middleware/retry/v1";
import RetryMiddleware from "mappersmith/middleware/retry/v2";
import BasicAuth from "mappersmith/middleware/basic-auth";
import CSRF from "mappersmith/middleware/csrf";
import Duration from "mappersmith/middleware/duration";
import EncodeJSON from "mappersmith/middleware/encode-json";
import ErrorHandler from "mappersmith/middleware/global-error-handler";
import Log from "mappersmith/middleware/log";
import Timeout from "mappersmith/middleware/timeout";

configs.gateway = FetchGateway;
configs.fetch = fetch as any;

export const github = forge({
  clientId: "github",
  middleware: [
    RetryMiddleware(),
    BasicAuth({
      username: "username",
      password: "password",
    }),
    CSRF(),
    Duration,
    EncodeJSON,
    ErrorHandler,
    Log,
    Timeout(1000),
  ],
  host: "https://www.githubstatus.com",
  resources: {
    Status: {
      current: { path: "/api/v2/status.json" },
      summary: { path: "/api/v2/summary.json" },
      components: { path: "/api/v2/components.json" },
    },
  },
});
