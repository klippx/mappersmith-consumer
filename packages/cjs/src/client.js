const forge = require("mappersmith").default;
const { configs } = require("mappersmith");
const FetchGateway = require("mappersmith/gateway/fetch").default;
const _HTTPGateway = require("mappersmith/gateway/http");
const _MockGateway = require("mappersmith/gateway/mock");
// const _XHRGateway = require("mappersmith/gateway/xhr");

// const _RetryMiddlewareV1 = require("mappersmith/middleware/retry");
const _RetryMiddlewareV1Alt = require("mappersmith/middleware/retry/v1");
const RetryMiddleware = require("mappersmith/middleware/retry/v2");
const BasicAuth = require("mappersmith/middlewares/basic-auth");
const CSRF = require("mappersmith/middleware/csrf");
const Duration = require("mappersmith/middleware/duration");
const EncodeJSON = require("mappersmith/middleware/encode-json");
const ErrorHandler = require("mappersmith/middleware/global-error-handler");
const Log = require("mappersmith/middleware/log");
const Timeout = require("mappersmith/middleware/timeout");

configs.gateway = FetchGateway;

const github = forge({
  clientId: "github",
  middleware: [
    RetryMiddleware.default(),
    BasicAuth.default({
      username: "username",
      password: "password",
    }),
    CSRF.default(),
    Duration.default,
    EncodeJSON.default,
    ErrorHandler.default,
    Log.default,
    Timeout.default(1000),
  ],
  host: "https://www.githubstatus.com",
  resources: {
    Status: {
      current: { path: "/api/v2/status.json" },
      summary: { path: "/api/v2/summary.json" },
      components: { path: "/api/v2/components.json" },
    },
  },
});

module.exports = github;
