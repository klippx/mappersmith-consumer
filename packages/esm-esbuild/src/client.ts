import { forge, configs } from "mappersmith";
import {
  FetchGateway,
  HTTPGateway as _HTTPGateway,
  MockGateway as _MockGateway,
  XHRGateway as _XHRGateway,
} from "mappersmith/gateway";

import {
  RetryMiddleware,
  BasicAuthMiddleware,
  CsrfMiddleware,
  DurationMiddleware,
  EncodeJsonMiddleware,
  GlobalErrorHandlerMiddleware,
  LogMiddleware,
  TimeoutMiddleware,
} from "mappersmith/middleware";

configs.gateway = FetchGateway;

export const github = forge({
  clientId: "github",
  middleware: [
    RetryMiddleware(),
    BasicAuthMiddleware({
      username: "username",
      password: "password",
    }),
    CsrfMiddleware(),
    DurationMiddleware,
    EncodeJsonMiddleware,
    GlobalErrorHandlerMiddleware,
    LogMiddleware,
    TimeoutMiddleware(1000),
  ],
  host: "https://www.githubstatus.com",
  resources: {
    Status: {
      current: { path: "/api/v2/status.json" },
      summary: { path: "/api/v2/summary.json" },
      components: { path: "/api/v2/components.json" },
    },
  },
});
