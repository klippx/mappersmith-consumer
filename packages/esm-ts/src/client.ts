import { forge, configs } from "mappersmith";
import {
  FetchGateway,
  HTTPGateway as _HTTPGateway,
  MockGateway as _MockGateway,
  XHRGateway as _XHRGateway,
} from "mappersmith/gateway";

import {
  RetryMiddleware,
  BasicAuthMiddleware,
  CsrfMiddleware,
  DurationMiddleware,
  EncodeJsonMiddleware,
  GlobalErrorHandlerMiddleware,
  LogMiddleware,
  TimeoutMiddleware,
} from "mappersmith/middleware";

import { setContext as _setContext } from "mappersmith";
import { MethodDescriptor as _MethodDescriptor } from "mappersmith";
import { Request as _Request } from "mappersmith";
import { Response as _Response } from "mappersmith";
import { version as _version } from "mappersmith";

configs.gateway = FetchGateway;

export const github = forge({
  clientId: "github",
  middleware: [
    RetryMiddleware(),
    BasicAuthMiddleware({
      username: "username",
      password: "password",
    }),
    CsrfMiddleware(),
    DurationMiddleware,
    EncodeJsonMiddleware,
    GlobalErrorHandlerMiddleware,
    LogMiddleware,
    TimeoutMiddleware(1000),
  ],
  host: "https://www.githubstatus.com",
  resources: {
    Status: {
      current: { path: "/api/v2/status.json" },
      summary: { path: "/api/v2/summary.json" },
      components: { path: "/api/v2/components.json" },
    },
  },
});
