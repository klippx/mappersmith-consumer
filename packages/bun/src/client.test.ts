import {
  describe,
  afterAll,
  beforeAll,
  beforeEach,
  expect,
  test,
} from "bun:test";
import { install, uninstall, mockRequest, MockAssert } from "mappersmith/test";
import { github } from "./client.js";

describe("github client works", () => {
  let mockSource: MockAssert;
  beforeAll(() => {
    install();
  });

  afterAll(() => {
    uninstall();
  });

  beforeEach(() => {
    mockSource = mockRequest({
      method: "get",
      url: "https://www.githubstatus.com/api/v2/status.json",
      response: {
        status: 200,
        body: {
          status: {
            description: "All Systems Operational",
          },
        },
      },
    });
  });

  test("github client works", async () => {
    const response = await github.Status.current();
    expect(response.data()).toMatchSnapshot();
    expect(mockSource.callsCount()).toEqual(1);
  });
});
