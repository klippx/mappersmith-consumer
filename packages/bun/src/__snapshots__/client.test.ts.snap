// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`github client works github client works: 
      {
        "status": {
          "description": "All Systems Operational",
        },
      }
     1`] = `
{
  "status": {
    "description": "All Systems Operational",
  },
}
`;

exports[`github client works github client works 1`] = `
{
  "status": {
    "description": "All Systems Operational",
  },
}
`;
